### ISSUE TITLE
gitlab_deploy_token moving state causes rpc error

### ISSUE THREAD

**Issue by zdelanofw** [gitlabhq/terraform-provider-gitlab#513]
### Terraform Version

v0.14.3

### Affected Resource(s)

- `gitlab_deploy_token`

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Debug / Panic Output

Errors are produced from terraform plan/apply.
See [the abbreviated terraform log](https://gist.github.com/zdelanofw/32e0772e4ef65f6c6c51fb08019472b0#file-error-log).

The final error at the end of the Terraform output reads:

```txt
Error: rpc error: code = Canceled desc = context canceled
Error: rpc error: code = Unavailable desc = transport is closing
```

### Expected Behavior

Terraform apply/plan should run as expected.

### Actual Behavior

Errors were produced (as shown in the debug/panic output).

### Steps to Reproduce

1. Have a local module in a `modules` directory in your Terraform workspace
    that creates a `gitlab_deploy_token` resource. We'll call this module
    `example-module` and the workspace `local-workspace` for the sake of this
    example.
2. Run `terraform apply`. This will create the token.
3. Copy `example-module` to another repository and push it to its remote
    origin.
4. In `local-workspace`, change the source of the module from `example-module`
    to the URL of the remote module. Both the remote module and
    `example-module` should have exactly the same code.
5. Run `terraform init`.
6. Run `terraform plan` or `terraform apply`.
7. Terraform will error out and show the output shown in the "debug/panic
    output" section.

Even if you run `terraform destroy -target='path_to.deploy_token'` and recreate
the token with the remote module, running `terraform apply`/`terraform plan`
again will still cause this error.

**Comment by armsnyder** [comment]
The bug appears to be a nil pointer dereference on this line: https://github.com/gitlabhq/terraform-provider-gitlab/blob/v3.4.0/gitlab/resource_gitlab_deploy_token.go#L154

**Comment by armsnyder** [comment]
Fixed via #510
